components/org.wso2.carbon.identity.api.server.configs/org.wso2.carbon.identity.api.server.configs.v1/src/main/java/org/wso2/carbon/identity/api/server/configs/v1/core/ServerConfigManagementService.java

```diff
@@ -767,30 +774,63 @@ private RequestPathAuthenticatorConfig getAuthenticatorById(RequestPathAuthentic
         if (config instanceof RequestPathAuthenticatorConfig) {
             authenticator.setType(Authenticator.TypeEnum.REQUEST_PATH);
+            authenticator.setDefinedBy(Authenticator.DefinedByEnum.SYSTEM);
         } else {
             authenticator.setType(Authenticator.TypeEnum.LOCAL);
+            if (AuthenticatorPropertyConstants.DefinedByType.SYSTEM == config.getDefinedByType()) {
+                authenticator.setDefinedBy(Authenticator.DefinedByEnum.SYSTEM);
+                List<AuthenticatorProperty> authenticatorProperties =
+                        Arrays.stream(config.getProperties()).map(propertyToExternal)
+                                .collect(Collectors.toList());
+                authenticator.setProperties(authenticatorProperties);
+            } else {
+                authenticator.setDefinedBy(Authenticator.DefinedByEnum.USER);
+                resolveEndpointConfiguration(authenticator, config);
+            }
         }
         String[] tags = config.getTags();
         if (ArrayUtils.isNotEmpty(tags)) {
             authenticator.setTags(Arrays.asList(tags));
         }
-        List<AuthenticatorProperty> authenticatorProperties =
-                Arrays.stream(config.getProperties()).map(propertyToExternal)
-                        .collect(Collectors.toList());
-        authenticator.setProperties(authenticatorProperties);
         return authenticator;
     }
 
+    private void resolveEndpointConfiguration(Authenticator authenticator, LocalAuthenticatorConfig config)
+            throws IdentityApplicationManagementServerException {
+
+        try {
+            UserDefinedLocalAuthenticatorConfig userDefinedConfig = (UserDefinedLocalAuthenticatorConfig) config;
+            EndpointConfig endpointConfig = userDefinedConfig.getEndpointConfig().getEndpointConfig();
+
+            AuthenticationType authenticationType = new AuthenticationType();
+            authenticationType.setType(AuthenticationType.TypeEnum.fromValue(endpointConfig
+                    .getAuthentication().getType().toString()));
+            Map<String, Object> authenticatorProperties = new HashMap<>();
+            for (AuthProperty prop: endpointConfig.getAuthentication().getProperties()) {
+                authenticatorProperties.put(prop.getName(), prop.getValue());
+            }
+            authenticationType.setProperties(authenticatorProperties);
+
+            Endpoint endpoint = new Endpoint();
+            endpoint.setAuthentication(authenticationType);
+            endpoint.setUri(userDefinedConfig.getEndpointConfig().getEndpointConfig().getUri());
+            authenticator.addEndpointItem(endpoint);
+        } catch (ClassCastException e) {
+            throw new IdentityApplicationManagementServerException("Error occurred while resolving endpoint " +
+                    "configuration of the authenticator.", e);
+        }
```

Comment:
Failure to cast this means this authenticator is not of user defined type isn't it ? Then it should rather be a client error IMO

Also is this related to endpoint or is this about line 812